### PR TITLE
Rootkeyevent

### PR DESCRIPTION
**Pull request by tznind**
Application has a global event `RootMouseEvent`.   This PR adds a corresponding `RootKeyEvent` for keypress events.

### CODE REVIEW

**Comment by BDisp** [comment]
You have to swap the arguments because of the warnings.

```cs
		[Fact]
		[AutoInitShutdown]
		public void Test_RootKeyEvent_Cancel ()
		{
			Application.RootKeyEvent += SuppressKey;

			var tf = new TextField ();

			Application.Top.Add (tf);
			Application.Begin (Application.Top);

			Application.Driver.SendKeys ('a', ConsoleKey.A, false, false, false);
			Assert.Equal ("a", tf.Text.ToString ());

			// SuppressKey suppresses the 'j' key
			Application.Driver.SendKeys ('j', ConsoleKey.A, false, false, false);
			Assert.Equal ("a", tf.Text.ToString ());

			Application.RootKeyEvent -= SuppressKey;

			// Now that the delegate has been removed we can type j again
			Application.Driver.SendKeys ('j', ConsoleKey.A, false, false, false);
			Assert.Equal ("aj", tf.Text.ToString ());
		}

		private bool SuppressKey (KeyEvent arg)
		{
			if (arg.KeyValue == 'j')
				return true;

			return false;
		}

```

**Comment by tznind** [comment]
Thanks, fixed.

File: Terminal.Gui/Core/Application.cs [hunk 1]
```diff
@@ -588,6 +591,15 @@ public static void UngrabMouse ()
 		/// </summary>
 		public static Action<MouseEvent> RootMouseEvent;
 
+		/// <summary>
+		/// <para>
+		/// Called for new KeyPress events before any processing is performed or
+		/// views evaluate.  Use for global key handling and/or debugging.
```
Comment (tig) [inline Terminal.Gui/Core/Application.cs:597]: Thank you for describing how it can be used here!

**Review by tig — APPROVED** [review]
LGTM!

**Pull request merged**